### PR TITLE
#166790052 view, upload and amend profile fixed

### PR DESCRIPTION
**Pull request by shaluchandwani**
#### What does this PR do?

Allow users to view and edit their profiles

#### Description of Tasks to be completed?

Allow a user to update their:

First name
Last name
Username
Email
Biography
Profile picture

#### How should this be manually tested?

After cloning the repo, cd into it, type npm run test. To run the application type npm start

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/166790052

#### Screenshots (if appropriate)
Screen Shot 2019-08-24 at 16 11 47
Screen Shot 2019-08-24 at 16 11 35
Screen Shot 2019-08-24 at 16 11 19
Screen Shot 2019-08-24 at 16 11 09
Screen Shot 2019-08-24 at 16 10 56

### CODE REVIEW

**Review by pngabo — APPROVED** [review]

**Review by Joe-Joseph — APPROVED** [review]

File: .env.example [hunk 1]
```diff
@@ -1,2 +1,2 @@
-BASE_URL=http://localhost:3000
-BACKEND_URL=https://ah-jawans-backend.herokuapp.com
+REACT_APP_BASE_URL=https://ah-jawans-backend.herokuapp.com
```
Comment (billkabanga) [inline .env.example]: @shaluchandwani The values of the env variables must not be exact, the `.env.example` should just tell what values are expected. This is a security vulnerability on the project. an example would be `BACKEND_URL=project_backend_url` and in the readme one should told to get the variables from the dev team.

File: src/__tests__/components/Profile/ProfileEditForm.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { getMaxListeners } from 'cluster';
+
+
+const props = {
+  editProfile: jest.fn(),
+  user: {},
+}
+describe('<ProfileEditForm />', () => {
+  test('renders without crashing', () => {
+    // const props = { userCredentials: user };
```
Comment (billkabanga) [inline src/__tests__/components/Profile/ProfileEditForm.test.js]: Please remove this line. You should not have commented code in your PR unless it's a comment. Do this for all other areas with the same issue.

File: src/__tests__/redux/actions/editProfile.test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        const payload = {
+            firstname: 'shaluVC',
+        };
+        const expectedAction = {
+            type: userActionsTypes.EDIT_PROFILE_SUCCESS,
+            payload,
+        };
+        expect(patchUserAction(payload)).toEqual(expectedAction);
+    });
+});
```
Comment (billkabanga) [inline src/__tests__/redux/actions/editProfile.test.js]: Please add a new line here and other files with the same issue.

**Review by billkabanga — CHANGES_REQUESTED** [review]

File: src/components/Header/HeaderUserMenu/HeaderUserMenu.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+          {isAuth && (
+            <li>
+              <Link to="/profile">
+                <FontAwesomeIcon icon={faUserCircle} /> Profile
+              </Link>
+            </li>
+          )}
+
+          {/* {isAuth && (
```
Comment (billkabanga) [inline src/components/Header/HeaderUserMenu/HeaderUserMenu.js]: Please remove this commented out code.

File: src/components/Profile/ProfileUserDetails/ProfileUserDetails.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                </span>
+                <span className="following">
+                  <span className="number">{0}</span> Following
+                </span>
+              </div>
+              <div className="divider" />
+              <span className="inline-block medium-v-padding">
+                <ProfileEdit />
+              </span>{' '}
+              {/* <span className="inline-block medium-v-padding">
```
Comment (billkabanga) [inline src/components/Profile/ProfileUserDetails/ProfileUserDetails.js]: The same here, remove the commented out code

**Review by billkabanga — CHANGES_REQUESTED** [review]

**Pull request merged**